packages/atomic/cypress/integration/result-list/result-components/result-html.cypress.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  generateComponentHTML,
+  TagProps,
+  TestFixture,
+} from '../../../fixtures/test-fixture';
+import * as CommonAssertions from '../../common-assertions';
+import {addResultList, buildTemplateWithSections} from '../result-list-actions';
+import {
+  resultHtmlComponent,
+  ResultHtmlSelectors,
+} from './result-html-selectors';
+
+interface ResultHtmlProps {
+  field?: string;
+  sanitize?: boolean;
+}
+
+const addResultHTMLInResultList = (props: ResultHtmlProps = {}) =>
+  addResultList(
+    buildTemplateWithSections({
+      bottomMetadata: generateComponentHTML(
+        resultHtmlComponent,
+        props as TagProps
+      ),
+    })
+  );
+
+describe('Result Html Component', () => {
+  describe('when not used inside a result template', () => {
+    beforeEach(() => {
+      new TestFixture()
+        .withElement(generateComponentHTML(resultHtmlComponent))
+        .init();
+    });
+
+    CommonAssertions.assertRemovesComponent(ResultHtmlSelectors.shadow);
+    CommonAssertions.assertConsoleError();
+  });
+
+  describe('when the field does not exist for the result', () => {
+    beforeEach(() => {
+      new TestFixture()
+        .with(addResultHTMLInResultList({field: 'thisfielddoesnotexist'}))
+        .init();
+    });
+
+    CommonAssertions.assertRemovesComponent(ResultHtmlSelectors.firstInResult);
+  });
+
+  describe('when the field value is not a string', () => {
+    const field = 'hello';
+    beforeEach(() => {
+      new TestFixture()
+        .with(addResultHTMLInResultList({field}))
+        .withCustomResponse((response) =>
+          response.results.forEach((result) => (result.raw[field] = 1337))
+        )
+        .init();
+    });
+
+    CommonAssertions.assertRemovesComponent(ResultHtmlSelectors.firstInResult);
+    CommonAssertions.assertConsoleError(false);
+  });
+
+  describe('when the field value exists & is an HTML string', () => {
+    const field = 'hello_world';
+    const rawValue = '<img src="google.com" onerror="console.log()" />';
+    const sanitizedValue = '<img src="google.com" />';
+    function setupExistingFieldValue(sanitize: boolean) {
+      new TestFixture()
+        .with(
+          addResultHTMLInResultList({
+            field: field,
+            sanitize,
+          })
+        )
+        .withCustomResponse((response) =>
+          response.results.forEach((result) => {
+            result.raw[field] = rawValue;
+          })
+        )
+        .init();
+    }
+
+    describe('when the "sanitize" prop is true', () => {
```

Comment:
Would it be possible to implement some sort of unit test to see if the html is getting properly sanitized?

Reply:
Yes, this is what this test does in fact:  It verifies that it sanitize the `onerror` property of the image on line 67